component/src/test/java/org/wso2/extension/siddhi/io/file/FileSourceTestCase.java

```diff
@@ -0,0 +1,266 @@
+                        case 8:
+                            AssertJUnit.assertEquals("wso2", event.getData(0));
+                            break;
+                    }
+                }
+            }
+        });
+
+        siddhiAppRuntime.start();
+
+        Thread.sleep(1000);
+
+        //assert event count
+        AssertJUnit.assertEquals("Number of events", 8, count.get());
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void siddhiIoFileTest2() throws InterruptedException {
+        log.info("test SiddhiIoFile 2");
+        String streams = "" +
+                "@App:name('TestSiddhiApp')" +
+                "@source(type='file',mode='line', action.after.process='none'," +
+                "file.uri='" + dir + "/line/logs.txt'," +
+                "file.polling.interval='1000', " +
+                "@map(type='json'))" +
+                "define stream FooStream (symbol string, price float, volume long); " +
+                "define stream BarStream (symbol string, price float, volume long); ";
+
+        String query = "" +
+                "from FooStream " +
+                "select * " +
+                "insert into BarStream; ";
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+
+        siddhiAppRuntime.addCallback("BarStream", new StreamCallback() {
+
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                int n = count.incrementAndGet();
+
+                for (Event event : events) {
+                    switch (n) {
+                        case 1:
+                            AssertJUnit.assertEquals(0f, event.getData(1));
+                            break;
+                        case 2:
+                            AssertJUnit.assertEquals(1f, event.getData(1));
+                            break;
+                        case 3:
+                            AssertJUnit.assertEquals(2f, event.getData(1));
+                            break;
+                        case 4:
+                            AssertJUnit.assertEquals(3f, event.getData(1));
+                            break;
+                        case 5:
+                            AssertJUnit.assertEquals(4f, event.getData(1));
+                            break;
+                        case 10:
+                            AssertJUnit.assertEquals(9f, event.getData(1));
+                            break;
+                        case 12:
+                            AssertJUnit.assertEquals(11f, event.getData(1));
+                            break;
+                        case 15:
+                            AssertJUnit.assertEquals(14f, event.getData(1));
+                            break;
+                    }
+                }
+            }
+        });
+
+        siddhiAppRuntime.start();
+
+        Thread.sleep(3000);
```

Comment:
You should ideally improve this to use SiddhiTestHelper.waitForEvents(). Refer https://github.com/wso2-extensions/siddhi-io-http/blob/master/component/src/test/java/org/wso2/extension/siddhi/io/http/source/HttpBasicTests.java#L115 Make a note and attend later.

Reply:
Fixed with PR #10